backend/tracim_backend/lib/collaborative_document_edition/collaboration_document_edition.py

```diff
@@ -0,0 +1,124 @@
+)
+from tracim_backend.lib.collaborative_document_edition.models import (
+    CollaborativeDocumentEditionFileType,
+)
+from tracim_backend.lib.collaborative_document_edition.models import (
+    CollaborativeDocumentEditionToken,
+)
+from tracim_backend.lib.collaborative_document_edition.models import FileTemplateList
+from tracim_backend.lib.core.content import ContentApi
+from tracim_backend.lib.utils.utils import is_dir_exist
+from tracim_backend.lib.utils.utils import is_dir_readable
+from tracim_backend.lib.utils.utils import is_file_exist
+from tracim_backend.lib.utils.utils import is_file_readable
+from tracim_backend.models.auth import User
+from tracim_backend.models.data import Content
+
+
+class CollaborativeDocumentEditionApi(object):
+    def __init__(self, current_user: typing.Optional[User], session: Session, config: CFG) -> None:
+        self._session = session
+        self._user = current_user
+        self._config = config
+
+    def discover(self) -> typing.List[CollaborativeDocumentEditionFileType]:
+        response = requests.get(
+            self._config.COLLABORATIVE_DOCUMENT_EDITION__COLLABORA__BASE_URL + "/hosting/discovery",
+            timeout=2,
+        )
+        root = ElementTree.fromstring(response.text)
+        supported_collabora_file = []  # type: typing.List[CollaborativeDocumentEditionFileType]
+        for xml_app in root.findall("net-zone/app"):
+            mimetype = xml_app.get("name")
+            # INFO - G.M - 2019-07-17 - this list return also a non mimetype type,
+            # we should not add him.
+            if mimetype == "Capabilities":
+                continue
+            for xml_action in xml_app:
+                extension = xml_action.get("ext")
+                url_source = xml_action.get("urlsrc")
+                associated_action = xml_action.get("name")
+                supported_collabora_file.append(
+                    CollaborativeDocumentEditionFileType(
+                        mimetype=mimetype,
+                        extension=extension,
+                        associated_action=associated_action,
+                        url_source=url_source,
+                    )
+                )
+        return supported_collabora_file
+
+    def get_collaboration_document_edition_config(self) -> CollaborativeDocumentEditionConfig:
+        return CollaborativeDocumentEditionConfig(
+            software=self._config.COLLABORATIVE_DOCUMENT_EDITION__SOFTWARE,
+            supported_file_types=self.discover(),
+        )
+
+    def get_file_template_list(self) -> FileTemplateList:
+        return FileTemplateList(file_templates=self._get_template_list())
+
+    def _has_template_extension(self, template_name: str, extension_list: typing.List[str]):
+        for extension in extension_list:
+            if template_name.endswith(extension):
+                return True
+        return False
+
+    def _get_template_list(self) -> typing.List[str]:
```

Comment:
what is a template? please put an example of value

Reply:
done.